[OV-65] feat(hub) : 허브 수정 서비스 구현

cnxw4570123 opened the pull request:
- Create 부분에서 값 받아오지 않는 것이 있어서 해당 부분 수정
- Update 관련 로직 추가
- Hub관련 예외 추가

## 변경 타입
- [x] 신규 기능 추가/수정
- [ ] 버그 수정
- [ ] 리팩토링
- [ ] 설정
- [ ] 비기능 (주석 등 기능에 영향을 주지 않음)

## 테스트 항목
- [x] 코드가 의도한 대로 동작하는지 테스트 완료
- [ ] 기존 기능에 영향을 주지 않음
- [ ] 관련 문서를 업데이트했거나 업데이트가 필요하지 않음

## 변경 내용
- **as-is**
    - CreateRequestDto에 포함되지 않은 엔티티 필드 존재.

- **to-be**
    - 필요한 CreateRequestDto에 필요한 필드 추가
    - Hub관련 예외 전용 클래스 추가
    - protected 메서드였던 BaseEntity의 생성, 수정, 삭제 마킹 메서드 public으로 변경
    - 업데이트 관련 로직을 엔티티 내부 기능으로 추가

## 참조
[OV-65](https://challduck.atlassian.net/jira/software/projects/OV/boards/3?selectedIssue=OV-65)

## 코멘트
- 구현하다가 수정할 부분 생각나서 넣다보니까.. 파일량이 많아졌습니다.. 다음부터는 잘게 쪼개서 올려놓을게요
- 죄송합니다ㅠ

cnxw4570123 committed:
[OV-65] feat(hub) : 허브 수정 서비스 구현

- Create 부분에서 값 받아오지 않는 것이 있어서 해당 부분 수정
- Update 관련 로직 추가
- Hub관련 예외 추가

cnxw4570123 committed:
Merge remote-tracking branch 'origin' into feat(hub)/implent-hub-service

challduck reviewed (approved):
LGTM!!

ggumi030 commented on hub/src/main/java/com/owlexpress/hub/domain/entity/Hub.java:
```diff
@@ -58,4 +62,34 @@ public Hub(UUID hubId, String name, String hubAddress, Point location, Long user
     }
+
+    public void update(HubUpdateRequestDto requestDto) {
+        Optional.ofNullable(requestDto.getName()).ifPresent(name -> this.name = name);
+
+        Optional.ofNullable(requestDto.getHubAddress())
+                .ifPresent(address -> this.hubAddress = address);
+
+        Optional.ofNullable(getLocation(requestDto))
+                .ifPresent(location -> this.location = location);
+
+        Optional.ofNullable(requestDto.getUserId()).ifPresent(userId -> this.userId = userId);
+
+        Optional.ofNullable(requestDto.getUserName())
+                .ifPresent(username -> this.userName = username);
+
+        Optional.ofNullable(requestDto.getUserPhoneNumber())
+                .ifPresent(phoneNumber -> this.userPhoneNumber = phoneNumber);
+
+        Optional.ofNullable(requestDto.getParentId())
+                .ifPresent(parentId -> this.parentHubId = parentId);
+    }
```
update 시에도 Null 체크를 해주시는군요 !! Null과의 싸움 !! 인상 깊게 봤습니다 

만약 RequestDto Valid에서 Null 체크를 다 해주면 이 Optional.ofNullable ~ 로직이 반복되지 않아도 괜찮지 않을까요 ? (다른 부분에서 발생할 수 있는지 궁금합니다)

cnxw4570123 replied:
앗.. 생각해보니까 PUT 매핑이라서 null 자체가 들어올 일이 없을 것 같네요..? 이 부분 수정해서 반영하도록 하겠습니다!

ggumi030 reviewed (approved):
고생하셨어요 !

Sunro1994 commented on hub/src/main/java/com/owlexpress/hub/presentation/dto/HubUpdateRequestDto.java:
```diff
@@ -0,0 +1,58 @@
+    @NotBlank(message = "유저 이름은 비어있을 수 없습니다.")
+    private String userName;
+
+    @Pattern(regexp = "^01(?:0|1|[6-9])-(?:\\d{3}|\\d{4})-\\d{4}$", message = "전화번호 형식이 일치하지 않습니다.")
```
전화번호도 디테일을 챙겨주셨군요...! 저도 적용해야겠어요 ㅋㅋㅋㅋㅋ 이 부분도 맞춰야겠네요

cnxw4570123 replied:
저도 미리 해놓지 않으면 까먹을 것 같았어요ㅋㅋㅋㅋ

Sunro1994 reviewed (approved):
LGTM!

Sunro1994 commented:
보는데 전혀 불편함은 없었습니다! 이정도 코드는 저는 거뜬해요!

cnxw4570123 committed:
[OV-65] refactor(hub) : requestDto 패키지 수정 및 Update null 체크 제거

- PUT 요청으로 들어오기 때문에 모든 필드가 채워진 상태로 들어오며, dto단에서 validate하기 때문에 null 체크하지 않도록 수정
- XXXRequestDto를 request 패키지로 이동
